### DISCUSSION TITLE
Support for Dart 3

### DISCUSSION THREAD

**Discussion by dukefirehawk** [dart-backend/angel#97] [General]
Version 8 will bump the min Dart SDK to 3

**Comment by DaneCryesSx** [comment]
Thanks for the update. It's good to know that the upcoming version 8 of the package will require a minimum Dart SDK of 3. This means that any projects using the package will also need to update their SDK version accordingly. It's always important to stay up to date with the latest SDK versions to ensure compatibility and take advantage of any new features and improvements.

**Reply by dukefirehawk** [reply]
Also replace the deprecated API with their replacement.

**Comment by GladisHosterXY** [comment]
Hey dukefirehawk!

It's great to hear that version 8 of the project will be bumping the minimum Dart SDK to 3. It's always exciting to see new features and improvements in programming languages.

Do you have any specific plans or goals for incorporating Dart 3 into the project? Are there any new features or changes in the language that you're particularly looking forward to using?

**Reply by dukefirehawk** [reply]
Primary goal is just to get the entire code base to work with Dart 3 then continue to refactor the code for better performance. Records and patterns look interesting and could be very useful for making ORM better.

**Comment by BerryBomarB14** [comment]
Exciting news for the upcoming Version 8! We're bumping up the minimum Dart SDK to 3.

**Comment by ChandaLoehrsE2** [comment]
This decision shows the Dart team's determination to push the programming language forward, and it may also bring some troubles and changes to developers who use older versions of the SDK.

**Comment by dukefirehawk** [comment]
Based on the porting work done so far, very minimal changes are required to move to Dart 3 unlike migrating from dart 2.10 to 2.12. Also a fair bit of code can be simplified. This I believe will make new features to be added in less time and some of the long standing bugs to be addressed. The 'belatuk' series of packages which are required by Angel3 has now been published as beta.1. Porting is starting soon for 8.0.0-beta.1 release. My prime beef is finding time to address  performance.
